File: dotnet/src/altgraph_shared_app/Services/Graph/v2/JRankComparator.cs
namespace altgraph_shared_app.Services.Graph.v2
{
  public class JRankComparator : IComparer<JRank>
  {
    public int Compare(JRank? x, JRank? y)
    {
      if (x == null)
      {
        return -1;
      }

      if (y == null)
      {
        return 1;
      }

      return x.CompareTo(y);
    }
  }
}

File: java/altgraph_web_app/src/main/java/org/cjoakim/cosmos/altgraph/data/common/graph/v2/JRankComparator.java
package org.cjoakim.cosmos.altgraph.data.common.graph.v2;

import java.util.Comparator;

/**
 * Comparator for sorting JRank objects.
 * <p>
 * Chris Joakim, Microsoft, November 2022
 */
public class JRankComparator implements Comparator {

    public int compare(Object obj1, Object obj2) {
        JRank pr1 = (JRank) obj1;
        JRank pr2 = (JRank) obj2;
        return pr1.compareTo(pr2);
    }
}
